mithril-aggregator/src/runtime/runner.rs

```diff
@@ -212,16 +188,24 @@ impl AggregatorRunnerTrait for AggregatorRunner {
 
         for signed_entity_type in signed_entity_types {
             if let Some(open_message) = self
-                .get_current_non_certified_open_message_for_signed_entity_type(&signed_entity_type)
+                .get_current_open_message_for_signed_entity_type(&signed_entity_type)
                 .await
-                .with_context(|| format!("AggregatorRunner can not get current non certified open message for signed entity type: '{}'", &signed_entity_type))?
+                .with_context(|| format!("AggregatorRunner can not get current open message for signed entity type: '{}'", &signed_entity_type))?
             {
-                return Ok(Some(open_message));
+                if !open_message.is_certified {
+                    return Ok(Some(open_message));
+                }
+                if open_message.is_certified || open_message.is_expired {
+                    continue;
+                }
             }
-            info!(
-                "RUNNER: get_current_non_certified_open_message: open message already certified";
-                "signed_entity_type" => ?signed_entity_type
-            );
+            let protocol_message = self.compute_protocol_message(&signed_entity_type).await.with_context(|| format!("AggregatorRunner can not compute protocol message for signed_entity_type: '{signed_entity_type}'"))?;
+
+            return Ok(Some(
+                        self.create_open_message(&signed_entity_type, &protocol_message)
```

Comment:
question: wouldn't it be more readable to create a binding with the `create_open_message` result and then return a `Ok(Some(result))` ?

Reply:
Done :heavy_check_mark: